modules/core/src/main/scala/com.snowplowanalytics.snowplow/scalatracker/Emitter.scala

```diff
@@ -32,13 +32,13 @@ trait ClockProvider[F[_]] {
  * This trait is needed here because `Tracker` is referentially transparent, and we leave it up for the emitters
  * to decide whether `F` will be referentially transparent also.
  */
-trait UUIDProvider[F[_]] {
+trait FUUIDProvider[F[_]] {
 
   /**
-   * Returns a random UUID. Most likely it will be `UUID.randomUUID()` wrapped in `F`
+   * Returns a random UUID. wrapped in `F`
    * @return current time in milliseconds
    */
-  def generateUUID: F[UUID]
+  def generateFUUID: F[FUUID]
 
 }
```

Comment:
You can completely remove `UUIDProvider`, since the `Sync` constraint on `F` lets us do `FUUID.randomFUUID[F]`.